### PR TITLE
feat(provider): handle local sign support in necessary methods

### PR DESCRIPTION
**Pull request by cbrzn**
this PR aims to add handle of signing for custom signers

### CODE REVIEW

File: provider/implementations/js/src/index.ts [hunk 1]
```diff
@@ -42,6 +42,69 @@ export class EthereumProviderPlugin extends Module<ProviderConfig> {
       return JSON.stringify("0x" + network.chainId.toString(16));
     }
 
+    if (args.method === "eth_getTransactionCount") {
+      try {
+        const signer = await connection.getSigner();
+        const nonce = await signer.getTransactionCount("latest");
```
[the remark this answers is not in the document]
Comment (dOrgJelli) [reply provider/implementations/js/src/index.ts:48]: I ended up just removing this, and instead think it's just best to issue an RPC call for this every time.

File: provider/implementations/js/src/index.ts [hunk 2]
```diff
@@ -42,6 +42,69 @@ export class EthereumProviderPlugin extends Module<ProviderConfig> {
...
+        return JSON.stringify("0x" + nonce.toString(16));
+      } catch (_) {
+        return "0x0";
+      }
+    }
+
+    if (args.method === "eth_sendTransaction") {
+      if (params.length < 1) {
+        throw new Error(
+          "Parameters must be given in eth_sendTransaction method"
```
[the remark this answers is not in the document]
Comment (dOrgJelli) [reply provider/implementations/js/src/index.ts]: fixed.

File: provider/implementations/js/src/index.ts [hunk 3]
```diff
@@ -42,6 +42,69 @@ export class EthereumProviderPlugin extends Module<ProviderConfig> {
...
+        );
+      }
+
+      const [transaction] = params;
+      if (connection.getSignerType() == SignerType.CUSTOM_SIGNER) {
+        const signer = await connection.getSigner();
+
+        // Ethers.js expects "0" | "1" | "2"
+        // but it's being received as hex (e.g: "0x02")
+        if ("type" in transaction) {
```
[the remark this answers is not in the document]
Comment (dOrgJelli) [reply provider/implementations/js/src/index.ts]: fixed.

**Review by dOrgJelli — APPROVED** [review]

File: provider/implementations/js/src/index.ts [hunk 4]
```diff
@@ -42,6 +42,69 @@ export class EthereumProviderPlugin extends Module<ProviderConfig> {
...
+      if (params.length < 2) {
+        throw new Error(
+          "Address & TypedData must be given in eth_signTypedData method"
+        );
+      }
+      if (connection.getSignerType() == SignerType.CUSTOM_SIGNER) {
+        const signer = await connection.getSigner()
+        let signature = "";
+        // This is a hack because in ethers v5.7 this method is experimental
+        // when when we update to ethers v6 this wont be needed. More info:
```
[the remark this answers is not in the document]
Comment (cbrzn) [reply provider/implementations/js/src/index.ts]: nice q... nothing tbh

**Pull request merged**